layers/cmd_buffer_state.cpp

```diff
@@ -1041,6 +1038,73 @@ void CMD_BUFFER_STATE::UpdateLastBoundDescriptorSets(VkPipelineBindPoint pipelin
     }
 }
 
+void CMD_BUFFER_STATE::UpdateLastBoundDescriptorBuffers(VkPipelineBindPoint pipeline_bind_point,
+                                                        const PIPELINE_LAYOUT_STATE &pipeline_layout, uint32_t first_set,
+                                                        uint32_t set_count, const uint32_t *buffer_indicies,
+                                                        const VkDeviceSize *buffer_offsets) {
+    uint32_t required_size = first_set + set_count;
+    const uint32_t last_binding_index = required_size - 1;
+    assert(last_binding_index < pipeline_layout.set_compat_ids.size());
+
+    // Some useful shorthand
+    const auto lv_bind_point = ConvertToLvlBindPoint(pipeline_bind_point);
+    auto &last_bound = lastBound[lv_bind_point];
+    last_bound.pipeline_layout = pipeline_layout.layout();
+    auto &pipe_compat_ids = pipeline_layout.set_compat_ids;
+    // Resize binding arrays
+    uint32_t last_set_index = first_set + set_count - 1;
+    if (last_set_index >= last_bound.per_set.size()) {
+        last_bound.per_set.resize(last_set_index + 1);
+    }
```

Comment:
Why is `last_set_index` needed?

```suggestion
    // Resize binding arrays
    if (last_binding_index >= last_bound.per_set.size()) {
        last_bound.per_set.resize(required_size);
    }
```